src/features/controls.ts

```diff
@@ -246,39 +261,38 @@ export const ControlsMixin = (ModelViewerElement:
          * state.
          */
         [$updateCamera]() {
-          const camera = this[$camera];
-          const controls = this[$controls];
           const scene = (this as any)[$scene];
+          const controls = this[$controls];
+          const framedHeight = scene.framedHeight;
 
-          camera.fov = this[$fov];
-          let near = (scene.framedHeight / 2) /
-              Math.tan((camera.fov / 2) * Math.PI / 180);
-          camera.near = scene.framedHeight / 10.0;
-          camera.far = scene.framedHeight * 10.0;
+          // Make zoom sensitivity scale with model size:
+          const zoomSensitivity = framedHeight / 10;
+          const framedDistance = (framedHeight / 2) /
+              Math.tan((controls.getCameraFov() / 2) * Math.PI / 180);
+          const near = framedHeight / 10.0;
+          const far = framedHeight * 10.0;
 
           // When we update the idealCameraDistance due to reframing, we want to
           // maintain the user's zoom level (how they have changed the camera
           // radius), which we represent here as a ratio.
-          const zoom = controls ? controls.getCameraSpherical().radius /
+          const zoom = this[$idealCameraDistance] > 0 ?
+              controls.getCameraSpherical().radius /
                   this[$idealCameraDistance] :
-                                  1;
-          this[$idealCameraDistance] = near + scene.modelDepth / 2;
+              1;
+          this[$idealCameraDistance] = framedDistance + scene.modelDepth / 2;
 
-          camera.aspect = scene.aspect;
-          camera.updateProjectionMatrix();
+          controls.updateIntrinsics(near, far, scene.aspect, zoomSensitivity);
 
           // Zooming out beyond the 'frame' doesn't serve much purpose
           // and will only end up showing the skysphere if zoomed out enough
-          const minimumRadius = camera.near + scene.framedHeight / 2.0;
+          const minimumRadius = near + framedHeight / 2.0;
           const maximumRadius = this[$idealCameraDistance];
 
           controls.applyOptions({minimumRadius, maximumRadius});
-          controls.updateFramedHeight(scene.framedHeight);
-
-          controls.target.copy(scene.target);
 
           controls.setRadius(zoom * this[$idealCameraDistance]);
-          controls.jumpToDestination();
+          controls.setTarget(scene.target);
+          controls.jumpToGoal();
```

Comment:
We previously agreed to use `Destination` as the domain term here. What changed?

Reply:
I had a pile of variables called "destX" and you told me not to use shorthand, so instead of making every variable super long, I switched from destination to goal. I switched this function for consistency (along with the comments).

Reply:
The new name works fine. The justification for making the change is one that I would prefer to avoid in the future. My general opinion on names is that code clarity trumps the cost of typing, and also the perceived byte cost (tools will fix this for us). In this case, both words are perfectly clear, but previously we had agreed to use the former word, and consensus should also be weighed in the balance 👍 

No need to revert, so long live `Goal`!

Reply:
Understood, will keep in mind in future.